tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -918,11 +924,66 @@ public function it_should_create_mime_types_for_allowed_sizes_only_via_global_va
 		$this->assertImageNotHasSizeSource( $attachment_id, 'not_allowed_size_200x150', 'image/webp' );
 	}
 
+	/**
+	 * Test image quality for image conversion.
+	 *
+	 * @test
+	 */
+	public function it_should_set_quality_with_image_conversion() {
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png', array( 'mime_type' => 'image/png' ) );
+
+		// Quality setting for the source image. For PNG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for PNG is 82.' );
+
+		// Temporary file.
+		$file = wp_tempnam();
+
+		// A PNG image will be converted to WebP whose quality should be 82 universally.
+		$editor->save( $file, 'image/webp' );
+		$this->assertSame( 82, $editor->get_quality(), 'Output image format is WebP. Quality setting for it should be 82 universally.' );
+
+		$editor = wp_get_image_editor( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+
+		// Quality setting for the source image. For JPG the fallback default of 82 is used.
+		$this->assertSame( 82, $editor->get_quality(), 'Default quality setting for JPG is 82.' );
+
+		// A JPG image will be converted to WebP whose quality should be 82 universally.
+		$editor->save( $file, 'image/webp' );
+		$this->assertSame( 82, $editor->get_quality(), 'Output image format is WebP. Quality setting for it should be 82 universally.' );
+
+		// Delete the temporary file.
+		unlink( $file );
+	}
+
+	/**
+	 * Test webp_uploads_modify_webp_quality function for image quality.
+	 *
+	 * @covers ::webp_uploads_modify_webp_quality()
+	 *
+	 * @test
+	 */
+	public function it_should_return_correct_quality_for_mime_types() {
+		$this->assertSame( 82, webp_uploads_modify_webp_quality( 90, 'image/webp' ), 'WebP image quality should always be 82.' );
+		$this->assertSame( 82, webp_uploads_modify_webp_quality( 82, 'image/webp' ), 'WebP image quality should always be 82.' );
+		$this->assertSame( 80, webp_uploads_modify_webp_quality( 80, 'image/jpeg' ), 'JPEG image quality should return default quality provided from WP filter wp_editor_set_quality.' );
+		$this->assertNotSame( 50, webp_uploads_modify_webp_quality( 40, 'image/jpeg' ), 'JPEG image quality should return default quality provided from WP filter wp_editor_set_quality.' );
+	}
+
 	/**
 	 * Runs (empty) hooks to satisfy webp_uploads_in_frontend_body() conditions.
 	 */
 	private function mock_frontend_body_hooks() {
 		remove_all_actions( 'template_redirect' );
 		do_action( 'template_redirect' );
 	}
+
+	/**
+	 * Return WebP image quality 86 for testing.
+	 */
+	public function return_webp_image_quality( $quality, $mime_type ) {
```

Comment:
Let's call this something like `force_high_webp_quality`. The current name doesn't really describe the purpose of this filter method.

Reply:
@felixarntz @aaemnnosttv We don't need to change quality for all the test cases here. Actually, the two test cases which are failing are both special/different.

1. it_should_create_webp_for_full_size_which_is_smaller_in_webp_format - this test case requires only full size image to generate WebP image for sources, and all other sub sizes should not generate WebP sizes. so when we pass 90 as WebP quality it will generate WebP for all and it will fail there. so we need 86 here for WebP for full size image only.
2. it_should_create_webp_for_some_sizes_which_are_smaller_in_webp_format - this test case requires sub sizes to generate WebP image as sub sources, and full size will not generate WebP. so when we pass 90 as WebP quality it will generate WebP for all and it will fail there. so we need 86 here for WebP for sub sizes image only.

It is a kind of special case where we can't just force WebP to any higher quality as the test image used there is specifically used there.

I have made some specific changes to only force return WebP quality for these two test cases.